core/shared/src/main/scala/com/softwaremill/sttp/SttpBackendOptions.scala

```diff
@@ -56,18 +74,21 @@ object SttpBackendOptions {
     Empty.socksProxy(host, port)
 
   private def loadSystemProxy: Option[Proxy] = {
-    def system(hostProp: String, portProp: String, make: (String, Int) => Proxy, defaultPort: Int) = {
+    def system(hostProp: String, portProp: String, nonProxyHostsPropOption: Option[String], make: (String, Int, List[String]) => Proxy, defaultPort: Int) = {
       val host = Option(System.getProperty(hostProp))
       def port = Try(System.getProperty(portProp).toInt).getOrElse(defaultPort)
-      host.map(make(_, port))
+      def nonProxyHosts: List[String] = nonProxyHostsPropOption
+        .flatMap(nonProxyHostsProp => Try(System.getProperty(nonProxyHostsProp).split("|").toList).toOption)
```

Comment:
The `|` syntax of separating non-proxy hosts is somehow standard?

Reply:
According to this doc : https://docs.oracle.com/javase/8/docs/api/java/net/doc-files/net-properties.html

Reply:
I just realised that it also allows pattern matching of hosts.